LEGO1/lego/legoomni/src/actors/act2actor.cpp

```diff
@@ -115,11 +138,197 @@ MxResult Act2Actor::VTable0x9c()
+			MxMatrix brickstrMatrix = brickstrROI->GetLocal2World();
+			brickstrMatrix[3][1] += 3.0f;
+			brickstrROI->UpdateTransformationRelativeToParent(brickstrMatrix);
+#endif
+			return;
+		}
+	}
+
+	if (!m_grec) {
+		if (m_unk0x1e == 2) {
+			m_unk0x1e = 0;
+			m_unk0x2c = m_shootAnim->GetDuration() + p_time;
+			m_unk0x30 = m_unk0x2c - 1300.0f;
+			SetWorldSpeed(0);
+			m_unk0x1c = FALSE;
+		}
+		else if (m_unk0x1e == 1) {
+			FindROI("pwrbrik")->SetVisibility(FALSE);
+			FindROI("debrick")->SetVisibility(FALSE);
+			FindROI("ray")->SetVisibility(FALSE);
+			m_unk0x4c = 0;
+			m_unk0x1e = 2;
+			VTable0xa0();
+			FUN_10019250(m_unk0x28 + 3, p_time + 3000.0f);
+		}
+		else if (m_unk0x1e == 0) {
+			if (m_unk0x40) {
+				m_unk0x40 = 0;
+				m_unk0x2c = m_shootAnim->GetDuration() + p_time;
+				m_unk0x30 = m_unk0x2c - 1300.0f;
+			}
+
+			if (FUN_10019700(p_time) == 1) {
+				return;
+			}
+		}
+		else if (m_unk0x1e == 5) {
+			FindROI("brickstr")->SetVisibility(FALSE);
+			GetROI()->SetVisibility(FALSE);
+			CurrentWorld()->RemoveActor(this);
+			return;
+		}
+#ifdef NDEBUG
+		else if (m_unk0x1e == 4) {
+			if (m_worldSpeed == 0.0f) {
+				return;
+			}
+
+			SetWorldSpeed(0.0f);
+			((LegoAct2*) CurrentWorld())->FUN_100517b0();
+			return;
+		}
+#endif
+	}
+
+	if (m_unk0x1e == 5 || m_unk0x1e == 4) {
+		return;
+	}
+
+	if (m_unk0x1e == 3) {
+		if (p_time - m_unk0x24 > 600.0f) {
+			m_unk0x1e = 2;
+			FUN_10019250(m_unk0x28 + 4, p_time + 15000.0f);
+		}
+	}
+	else {
+		roiPepper = FindROI("pepper");
+
+		if (roiPepper) {
+			vm = VideoManager()->Get3DManager()->GetLego3DView()->GetViewManager();
+			assert(vm);
+
+			MxU32 inFrustum = vm->IsBoundingBoxInFrustum(m_roi->GetWorldBoundingBox());
+
+			if (inFrustum) {
+				Mx3DPointFloat local18(roiPepper->GetWorldDirection());
+				Mx3DPointFloat local30(m_roi->GetWorldPosition());
+				Mx3DPointFloat local60(roiPepper->GetWorldPosition());
+				local30 -= local60;
+				local30.Unitize();
+
+				MxFloat dotproduct = local18.Dot(&local30, &local18);
+
+				if (dotproduct >= 0.0) {
+					pepperWorldPosition = roiPepper->GetWorldPosition();
+					worldPosition = m_roi->GetWorldPosition();
+
+					MxFloat distance1 = DISTSQRD3(pepperWorldPosition, worldPosition);
+
+					if (distance1 < 75.0f) {
+						if (!m_unk0x1c) {
+							m_unk0x1c = 1;
+
+							if (!m_unk0x1e) {
+								FUN_100199f0(2);
+								m_unk0x1e = 1;
+							}
+							else {
+								childROI = m_roi->FindChildROI("windsd", m_roi);
+								childPosition = childROI->GetWorldPosition();
+								distance2 = DISTSQRD3(pepperWorldPosition, childPosition);
+
+								childROI = m_roi->FindChildROI("reardr", m_roi);
+								childPosition = childROI->GetWorldPosition();
+								distance3 = DISTSQRD3(pepperWorldPosition, childPosition);
+
+								if (distance3 > distance2) {
+									FUN_100199f0(0);
+								}
+								else
+#ifdef NDEBUG
+									// `distance2` is guessed, looks promising
+									if (distance2 - m_unk0x24 > 3000.0f) {
```

Comment:
I'm fairly sure this should be `p_time` instead of `distance2`

Reply:
looks more promising!